src/pybind/rgw_admin_user/rgw_admin_user.pyx

```diff
@@ -0,0 +1,290 @@
+        prog_name = "rgw_admin_user.pyx"
+        cdef char* argv[1]
+        argv[0] = prog_name
+
+        PyEval_InitThreads()
+        self.state = ""
+        ret = librgw_admin_user_create(&self.handle, 1, argv)
+        if ret == 0:
+            self.state = "initialized"
+        else:
+            raise make_ex(ret, "error calling librgw_admin_user_create")
+
+    def shutdown(self):
+        """
+        Uninitialize/cleanup the library handle.
+        """
+        if self.state in ["initialized"]:
+            with nogil:
+                librgw_admin_user_shutdown(self.handle)
+            self.state = "shutdown"
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type_, value, traceback):
+        self.shutdown()
+
+    def __dealloc__(self):
+        self.shutdown()
+
+    def version(self):
+        """
+        Get the version number of the ``librgw_admin_user`` C library.
+
+        :returns: a tuple of ``(major, minor, extra)`` components of the
+                  librgw_admin_user version
+        """
+        cdef:
+            int major = 0
+            int minor = 0
+            int extra = 0
+        with nogil:
+            rgw_admin_user_version(&major, &minor, &extra)
+        return (major, minor, extra)
+
+    def create_user(self, uid, display_name, access_key, secret_key, email,
+                    caps, access, admin = False, system = False):
+        """
+        Create a user in the RGW cluster
+        """
+
+        #nogil discipline
+        uid = cstr(uid, 'uid')
+        display_name = cstr(display_name, '_display_name')
+        access_key = cstr(access_key, 'access_key')
+        secret_key = cstr(secret_key, 'secret_key')
+        email = cstr(email, 'email')
+        caps = cstr(caps, 'caps')
+        access = cstr(access, 'access')
+
+        cdef:
+            const_char *_uid = uid
+            const_char *_display_name = display_name
+            const_char *_access_key = access_key
+            const_char *_secret_key = secret_key
+            const_char *_email = email
+            const_char *_caps = caps
+            const_char *_access = access
+            bint _admin = admin
+            bint _system = system
+
+        with nogil:
+            ret = rgw_admin_create_user(self.handle, _uid, _display_name,
+                                        _access_key, _secret_key, _email,
+                                        _caps, _access, _admin, _system)
+        if ret < 0:
+            raise make_ex(ret, "error in user_create '%s'" % (uid))
+        return
+
+    def user_info(self, uid):
+        """
+        Get user information for `uid` in the RGW cluster
+        """
+
+        #nogil discipline
+        uid = cstr(uid, 'uid')
+
+        cdef:
+            const_char *_uid = uid
+            rgw_user_info user_info
+
+        with nogil:
+            ret = rgw_admin_user_info(self.handle, _uid, &user_info)
+
+        info = UserInfo(uid=user_info.uid,
```

Comment:
I'm missing the ``tenant`` field here. Working on the RGW part in the Dashboard showed me that this field is really important when you work with tenanted users and RGW Admin OPS API. The Admin OPS API is forcing you to generate a ``uid`` which contains the tenant name, see https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/controllers/rgw.py#L109 and  https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/controllers/rgw.py#L156.